modules/apps/headless-apio/blog/blog-apio-test/src/testIntegration/java/com/liferay/blog/apio/internal/architect/resource/test/BlogPostingNestedCollectionResourceTest.java

```diff
@@ -0,0 +1,195 @@
+
+		Assert.assertTrue(
+			"Journal articles: " + blogEntries,
+			blogEntries.contains(blogsEntry));
+	}
+
+	@Test
+	public void testGetPageItemsWith1Pending() throws Exception {
+		ServiceContext serviceContext = _getServiceContext(true);
+
+		BlogsEntry blogsEntry = _getBlogEntry(new Date(), serviceContext);
+
+		_blogsEntryLocalService.updateStatus(
+			serviceContext.getUserId(), blogsEntry.getEntryId(),
+			WorkflowConstants.STATUS_PENDING, serviceContext,
+			Collections.emptyMap());
+
+		int blogEntriesCount = _blogsEntryLocalService.getGroupEntriesCount(
+			_group.getGroupId(), _anyStatusQueryDefinition);
+
+		Assert.assertEquals(1, blogEntriesCount);
+
+		PageItems<BlogsEntry> pageItems = getPageItems(
+			PaginationRequest.of(10, 1), _group.getGroupId());
+
+		Assert.assertEquals(0, pageItems.getTotalCount());
+	}
+
+	@Test
+	public void testGetPageItemsWith1Scheduled() throws Exception {
+		ServiceContext serviceContext = _getServiceContext(true);
+
+		Date displayDate = new Date(
+			System.currentTimeMillis() + 24 * 60 * 60 * 1000);
+
+		_getBlogEntry(displayDate, serviceContext);
+
+		int blogEntriesCount = _blogsEntryLocalService.getGroupEntriesCount(
+			_group.getGroupId(), _anyStatusQueryDefinition);
+
+		Assert.assertEquals(1, blogEntriesCount);
+
+		PageItems<BlogsEntry> pageItems = getPageItems(
+			PaginationRequest.of(10, 1), _group.getGroupId());
+
+		Assert.assertEquals(0, pageItems.getTotalCount());
+	}
+
+	@Test
+	public void testGetPageItemsWithPermissions() throws Exception {
+		ServiceContext serviceContext = _getServiceContext(false);
+
+		_getBlogEntry(new Date(), serviceContext);
+
+		int blogEntriesCount = _blogsEntryLocalService.getGroupEntriesCount(
+			_group.getGroupId(), _anyStatusQueryDefinition);
+
+		Assert.assertEquals(1, blogEntriesCount);
+
+		User user = UserTestUtil.addUser();
+
+		PermissionChecker permissionChecker =
+			PermissionCheckerFactoryUtil.create(user);
+
+		try (ContextUserReplace contextUserReplace = new ContextUserReplace(
+				user, permissionChecker)) {
+
+			PageItems<BlogsEntry> pageItems = getPageItems(
+				PaginationRequest.of(10, 1), _group.getGroupId());
+
+			Assert.assertEquals(0, pageItems.getTotalCount());
+		}
+		finally {
+			_userLocalService.deleteUser(user);
+		}
+	}
+
+	private BlogsEntry _getBlogEntry(
+			Date displayDate, ServiceContext serviceContext)
+		throws PortalException {
+
+		return _blogsEntryLocalService.addEntry(
+			serviceContext.getUserId(), "headline", "alternativeHeadline",
+			"/friendly/url/path", "some description", "article body",
+			displayDate, true, true, new String[0], "image caption", null, null,
+			serviceContext);
+	}
+
+	private ServiceContext _getServiceContext(boolean permissionsEnabled)
+		throws PortalException {
+
+		ServiceContext serviceContext =
+			ServiceContextTestUtil.getServiceContext(_group.getGroupId());
+
+		serviceContext.setAddGuestPermissions(permissionsEnabled);
```

Comment:
Better if we don't encapsulate this logic.

It is better if we read it in the test, that read _getServiceContext(true) or _getServiceContext(false)